### DISCUSSION TITLE
Multiple configs with cypress 10.0

### DISCUSSION THREAD

**Discussion by techshashank80** [cypress-io/cypress#23371] [General]
I have multiple configs for different env in JSON format.  then how we can achieve same thing with cypress 10. As per cypress migration video they did not talk about to migrate these config plans. any clue?

**Comment by reggZez** [comment]
I'm still working with multiple configs, helps me with more envs. What I did is that I have changed my exampleEnvConfig.json (lets say it's for test env) to exampleEnvConfig.js, and inside of it I have the same code as described in cypress.config.js (added some settings example here for e2e)

![image](https://user-images.githubusercontent.com/111056328/184940344-6490fbe5-ac2b-4f7c-ac52-4907272399c8.png)

Later in the package.json I just specify in the script which config file should be opened when running that specific script, again lets say we are targeting test env (--config-file cypress/config/exampleEnvConfig.js)

![image](https://user-images.githubusercontent.com/111056328/184941364-c1ec740d-53dc-4f2d-864b-ca2d5c1fd93a.png)

When you run "npm run test" it will execute this script by opening cypress runner and using specified config.js.

If you run "npx cypress open"  it will run with cypress.config.js. I don't really use it but you can put some default config there in case someone runs without script.

Dunno if this helps you, maybe someone has better approach but this works for me.

**Reply by techshashank80** [reply]
Yes, this solution worked for me. Thank you, Appreciate it.

**Comment by lmiller1990** [comment]
Since it's just JS now you could also do something like:

```ts
import fs from 'fs-extra'

const config1 = fs.readJsonSync('./config1.json')
const config2 = fs.readJsonSync('./config12json')

export default defineConfig(
  process.env.USE_CONFIG_1 ? config1 : config2
)
```

Just an idea!

**Reply by techshashank80** [reply]
Thank you for the idea, we are already using 'fs-extra' but question is, does cypress 10.0 support the 'configFile' option through cli? we have to pass config file name from command line.

**Reply by techshashank80** [reply]
as per cypress 

You may use the --config-file command line flag or the configFile [module API](https://docs.cypress.io/guides/guides/module-api) option to specify a .js or .ts file. JSON config files are no longer supported.

any input on this?

**Reply by lmiller1990** [reply]
I tried this out `yarn cypress run --component --config-file cypress.blah.ts` and it correctly read the `cypress.blah.ts` file. Is that not working for you? The API should be identical, you just pass a path to a JS module, not a JSON file.

**Reply by techshashank80** [reply]
Yes above —config-file is working fine , my question was for configFile where we were just passing the json file name without file extension before 10.0 version. But in your example, json files are being read.  Is this cypress run env —configFile = stage no longer supported ?

**Reply by lmiller1990** [reply]
Just to clarify are you referring to [this](https://docs.cypress.io/guides/guides/environment-variables#Option-2-cypress-env-json)? It looks like it should still work. To clarify you want to have a `cypress.env.json`  and have it merged with your `cypress.config` at runtime?

It might be good if you can provide a minimum repo if you think there is a bug.

```json
{ 
  baseUrl: "http://staging.com"
}
```
